hatch_pip_compile/plugin.py

```diff
@@ -173,24 +173,19 @@ def _pip_sync_cli(self) -> None:
         self.virtual_env.platform.check_command(cmd)
         if not self.dependencies:
             self._piptools_lock_file.unlink()
+        if not self.skip_install:
+            if self.dev_mode:
+                super().install_project_dev_mode()
+            else:
+                super().install_project()
 
-    def install_project(self):
-        """
-        Install the project the first time
-
-        The same implementation as `sync_dependencies`
-        due to the way `pip-sync` uninstalls our root package
-        """
-        self._hatch_pip_compile_install()
-
-    def install_project_dev_mode(self):
+    def _pip_install_cli(self) -> None:
         """
-        Install the project the first time in dev mode
-
-        The same implementation as `sync_dependencies`
-        due to the way `pip-sync` uninstalls our root package
+        Install the project with `pip`
         """
-        self._hatch_pip_compile_install()
+        args = ["--requirement", str(self._piptools_lock_file)]
```

Comment:
Maybe `--no-deps` arg should be added? As I understand, it always just works, and maybe it'll save some work for the resolution -- or for installing unexpected optional packages?

Reply:
Right- a tough call. Whether a lock file is not really installable or not really locked 😅

Reply:
`--no-deps` is very similar to `pip-sync`, just without the uninstalling part. I'm leaning towards leaving `--no-deps` out for the default implementation and enabling with a feature flag. What would your preferred behavior be?

Reply:
That's reasonable